src/ui/units/collections/components/CollectionPage/CollectionPage.tsx

```diff
@@ -338,6 +377,85 @@ export const CollectionPage = React.memo<Props>(
             }
         };
 
+        const onUpdateCheckbox = (
+            checked: boolean,
+            type: 'workbook' | 'collection',
+            entityId: string,
+        ) => {
+            setSelectedMap({
+                ...selectedMap,
+                [entityId]: {
+                    type,
+                    checked,
+                },
+            });
+
+            if (checked && !isOpenSelectionMode) {
+                setIsOpenSelectionMode(true);
+            }
+        };
+
+        const onSelectAll = (checked: boolean) => {
+            const selected: SelectedMap = {};
+
+            contentItems.forEach((item) => {
+                const isWorkbook = 'workbookId' in item;
+                const id = isWorkbook ? item.workbookId : item.collectionId;
+                const type = isWorkbook ? 'workbook' : 'collection';
+
+                selected[id] = {
+                    type,
+                    checked,
+                };
+            });
+
+            setSelectedMap({
+                ...selected,
+            });
+
+            if (checked && !isOpenSelectionMode) {
+                setIsOpenSelectionMode(true);
+            }
+        };
+
+        const setBatchAction = () => {
+            const workbookIds: string[] = [];
+            const collectionIds: string[] = [];
+
+            Object.keys(selectedMap).forEach((key) => {
+                const item = selectedMap[key];
+                if (item.checked) {
+                    if (item.type === 'workbook') {
+                        workbookIds.push(key);
+                    } else {
+                        collectionIds.push(key);
+                    }
+                }
+            });
+            dispatch(
+                openDialog({
+                    id: DIALOG_MOVE_COLLECTIONS_WORKBOOKS,
+                    props: {
+                        open: true,
+                        onApply: refreshContent,
+                        onClose: handeCloseMoveDialog,
```

Comment:
Let's add `initialParentId: curCollectionId,` to start the dialog in a current collection.